src/types/output.rs

```diff
@@ -120,12 +152,17 @@ impl OutputLayout {
         unsafe { OutputLayout { layout: wlr_output_layout_create() } }
     }
 
-    pub fn add_auto(&mut self, output: &mut Output) {
-        unsafe { wlr_output_layout_add_auto(self.layout, output.to_ptr()) }
+    pub unsafe fn to_ptr(&self) -> *mut wlr_output_layout {
+        self.layout
     }
 
-    pub unsafe fn as_ptr(&self) -> *mut wlr_output_layout {
-        self.layout
+    pub unsafe fn from_ptr(layout: *mut wlr_output_layout) -> Self {
+        OutputLayout { layout }
+    }
+
+    pub fn remove(&mut self, output: &mut Output) {
+        // TODO Double check we can pass any output in here and not cause unsafety
+        unsafe { wlr_output_layout_remove(self.layout, output.to_ptr()) }
     }
 }
 
```

Comment:
For the drop impl (just below this line) reclaim the user data so that the reference isn't leaked